### ISSUE TITLE
Running this project crashes the server when opening the page

### ISSUE THREAD

**Issue by beeman** [strongloop/loopback-example-pubsub#2]
As the title says running this project crashes the server with the error message below: 

```
$ node server/server.js
Browse your REST API at http://0.0.0.0:3000/explorer
Web server listening at: http://0.0.0.0:3000/
events.js:85
      throw er; // Unhandled 'error' event
            ^
Error: connect ECONNREFUSED
    at exports._errnoException (util.js:746:11)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1010:19)
```

**Comment by kennethlynne** [comment]
+1

**Comment by danielfbm** [comment]
+1

**Comment by vizo** [comment]
For me it works...
Node v0.12.7
Do you have any other server on 3000 port?

**Comment by vizo** [comment]
+1 When you open localhost:3000 ....

**Comment by superkhau** [comment]
The instructions were missing a few steps. Please clone again and let me know if you run into any issues.

1. Browserify was missing.
2. No instructions for running gulp was present in the readme file

**Comment by beeman** [comment]
@superkhau thanks for looking into this issue. 

It still fails with the same error in my shell. The complete output of all the steps I took can be found here https://gist.github.com/beeman/394626bd9cbefff740c1

Below is what I see when I open the project in my browser:

<img width="596" alt="screen shot 2015-08-19 at 10 18 29" src="https://cloud.githubusercontent.com/assets/36491/9351665/b683f1d2-465b-11e5-821f-96b0a34d550b.png">

**Comment by vizo** [comment]
@superkhau i did that all by my self ... but it was still the same ... 
If i add to server.js:
```javascript
process.on('uncaughtException', function (err) {
});
```
Everything works ....

I think the problem is that primus is trying to connect to some port, but server is not listening yet... or something like this ...

**Comment by superkhau** [comment]
@beeman @vizo I tried it again, I'm also running into the two issues you guys stated. Escalating.

@ritch This example is failing. We also need better instructions in the README.md. Do you know what's going on?

**Comment by beeman** [comment]
Confirmed that the instructions that @ritch added about installing `mosquitto` helped me to get a working system.

On OS X it takes a bit more to start `mosquitto`, the package installed with homebrew does not seem to add the binary to the path. I thus followed the instructions after installing the package:

```
$ brew install mosquitto
	<snip>

$ mosquitto
zsh: command not found: mosquitto

$ mosquitto -c /usr/local/etc/mosquitto/mosquitto.conf
zsh: command not found: mosquitto

$ ln -sfv /usr/local/opt/mosquitto/*.plist ~/Library/LaunchAgents
/Users/beeman/Library/LaunchAgents/homebrew.mxcl.mosquitto.plist -> /usr/local/opt/mosquitto/homebrew.mxcl.mosquitto.plist

$ launchctl load ~/Library/LaunchAgents/homebrew.mxcl.mosquitto.plist
```

**Comment by superkhau** [comment]
This has been fixed in #6. Please let me know if any of you guys still run into issues and thanks @ritch.

**Comment by vizo** [comment]
Seems all ok ...
